src/libsyntax/config.rs

```diff
@@ -250,6 +257,50 @@ impl<'a> StripUnconfigured<'a> {
     pub fn configure_stmt(&mut self, stmt: ast::Stmt) -> Option<ast::Stmt> {
         self.configure(stmt)
     }
+
+    pub fn configure_struct_expr_field(&mut self, field: ast::Field) -> Option<ast::Field> {
+        if !self.features.map(|features| features.struct_field_attributes).unwrap_or(true) {
+            if !field.attrs.is_empty() {
+                let mut err = feature_err(&self.sess,
+                                          "struct_field_attributes",
+                                          field.span,
+                                          GateIssue::Language,
+                                          "attributes on struct literal fields are unstable");
+                err.emit();
+            }
+        }
+
+        self.configure(field)
+    }
+
+    pub fn configure_struct_pat_field(&mut self, spanned: Spanned<ast::FieldPat>)
+                                      -> Option<Spanned<ast::FieldPat>> {
+        let Spanned { span, node } = spanned;
+        if !self.features.map(|features| features.struct_field_attributes).unwrap_or(true) {
+            if !node.attrs.is_empty() {
+                let mut err = feature_err(&self.sess,
+                                          "struct_field_attributes",
+                                          span,
+                                          GateIssue::Language,
+                                          "attributes on struct pattern fields are unstable");
+                err.emit();
+            }
+        }
+
+        self.configure(node).map(|node| Spanned { span: span, node: node })
+    }
+
+    pub fn configure_pat_kind(&mut self, pat_kind: ast::PatKind) -> ast::PatKind {
+        match pat_kind {
+            ast::PatKind::Struct(path, fields, etc) => {
+                let fields = fields.into_iter()
+                    .filter_map(|a| self.configure_struct_pat_field(a))
```

Comment:
I believe you could derive `HasAttrs` for `Spanned<ast::FieldPat>`, remove `configure_struct_pat_field`, and gated feature check in this closure.